lib/CachingReflector.php

```diff
@@ -75,7 +75,7 @@ public function getParamTypeHint(\ReflectionFunctionAbstract $function, \Reflect
             $paramCacheKey = self::CACHE_KEY_CLASSES . "{$lowClass}.{$lowMethod}.param-{$lowParam}";
         } else {
             $lowFunc = strtolower($function->name);
-            $paramCacheKey = ($lowFunc !== '{closure}')
```

Comment:
This match only works with closures defined outside of all namespace. It seems like the original intent was to exclude all closures, which makes sense.